vscode-dotnet-runtime-library/src/Acquisition/DotnetCoreAcquisitionWorker.ts

```diff
@@ -78,18 +77,21 @@ export class DotnetCoreAcquisitionWorker implements IDotnetCoreAcquisitionWorker
     private async acquireCore(version: string, installRuntime: boolean): Promise<string> {
         const installingVersions = this.context.extensionState.get<string[]>(this.installingVersionsKey, []);
         const partialInstall = installingVersions.indexOf(version) >= 0;
-        if (partialInstall) {
+        if (partialInstall && installRuntime) {
             // Partial install, we never updated our extension to no longer be 'installing'.
             // uninstall everything and then re-install.
             this.context.eventStream.post(new DotnetAcquisitionPartialInstallation(version));
 
-            await this.uninstall(version);
+            await this.uninstallRuntime(version);
+        } else if (partialInstall) {
+            this.context.eventStream.post(new DotnetAcquisitionPartialInstallation(version));
+            await this.uninstallAll();
         }
 
-        const dotnetInstallDir = this.getDotnetInstallDir(version);
+        const dotnetInstallDir = this.context.installDirectoryProvider.getDotnetInstallDir(version, this.installDir);
         const dotnetPath = path.join(dotnetInstallDir, this.dotnetExecutable);
 
-        if (fs.existsSync(dotnetPath)) {
+        if (this.context.installDirectoryProvider.isBundleInstalled(dotnetPath, version, this.context.extensionState, this.installingVersionsKey)) {
```

Comment:
Why does this need to be part of the provider? I'd imagine that the worker could get the install directory and then based off of that determine if it's installed something into it.